internal/pkg/aws/cloudwatchlogs/cloudwatchlogs.go

```diff
@@ -0,0 +1,173 @@
+func (s *Service) logStreams(logGroupName string) ([]*string, error) {
+	resp, err := s.cwlogs.DescribeLogStreams(&cloudwatchlogs.DescribeLogStreamsInput{
+		LogGroupName: aws.String(logGroupName),
+		Descending:   aws.Bool(true),
+		OrderBy:      aws.String(orderByLastEventTime),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("describe log streams of log group %s: %w", logGroupName, err)
+	}
+	if len(resp.LogStreams) == 0 {
+		return nil, fmt.Errorf("no log stream found in log group %s", logGroupName)
+	}
+	logStreamNames := make([]*string, len(resp.LogStreams))
+	for ind, logStream := range resp.LogStreams {
+		logStreamNames[ind] = logStream.LogStreamName
+	}
+	return logStreamNames, nil
+}
+
+// LogEvents returns an array of Cloudwatch Logs events.
+func (s *Service) LogEvents(logGroupName string, streamTokens map[string]*string, opts ...GetLogEventsOpts) (*LogEventsOutput, error) {
+	var events []*Event
+	var in *cloudwatchlogs.GetLogEventsInput
+	nextForwardTokens := make(map[string]*string)
+	logStreamNames, err := s.logStreams(logGroupName)
+	if err != nil {
+		return nil, err
+	}
+	for _, logStreamName := range logStreamNames {
+		in = &cloudwatchlogs.GetLogEventsInput{
+			LogGroupName:  aws.String(logGroupName),
+			LogStreamName: logStreamName,
+			NextToken:     streamTokens[*logStreamName],
+		}
+		for _, opt := range opts {
+			opt(in)
+		}
+		resp, err := s.cwlogs.GetLogEvents(in)
+		if err != nil {
+			return nil, fmt.Errorf("get log events of %s/%s: %w", logGroupName, *logStreamName, err)
+		}
+
+		for _, event := range resp.Events {
+			taskID, err := parseTaskID(*logStreamName)
+			if err != nil {
+				return nil, err
+			}
+			log := &Event{
+				TaskID:        taskID,
+				IngestionTime: aws.Int64Value(event.IngestionTime),
+				Message:       aws.StringValue(event.Message),
+				Timestamp:     aws.Int64Value(event.Timestamp),
+			}
+			events = append(events, log)
+		}
+
+		nextForwardTokens[*logStreamName] = resp.NextForwardToken
+	}
+	sort.SliceStable(events, func(i, j int) bool { return events[i].Timestamp < events[j].Timestamp })
+	var truncatedEvents []*Event
+	if len(events) >= int(*in.Limit) {
+		truncatedEvents = events[len(events)-int(*in.Limit):]
+	} else {
+		truncatedEvents = events
+	}
+	return &LogEventsOutput{
+		Events:     truncatedEvents,
+		NextTokens: nextForwardTokens,
+	}, nil
+}
+
+// LogGroupExists returns if a log group exists.
+func (s *Service) LogGroupExists(logGroupName string) (bool, error) {
+	_, err := s.cwlogs.DescribeLogStreams(&cloudwatchlogs.DescribeLogStreamsInput{
+		LogGroupName: aws.String(logGroupName),
+	})
+	if err == nil {
+		return true, nil
+	}
+	aerr, ok := err.(awserr.Error)
+	if !ok {
+		return false, err
+	}
+	if aerr.Code() != cloudwatchlogs.ErrCodeResourceNotFoundException {
+		return false, err
+	}
+	return false, nil
+}
+
+func parseTaskID(logStreamName string) (string, error) {
+	logStreamNameSplit := strings.Split(logStreamName, "/")
```

Comment:
Can we add a comment with an an example of a logStreamName so that the reader can make sense of the magic numbers `3` and `2`?